st3/sublime_lib/resource_path.py

```diff
@@ -11,18 +11,102 @@
 __all__ = ['ResourcePath']
 
 
-def get_resource_roots():
-    return {
-        'Packages': sublime.packages_path(),
-        'Cache': sublime.cache_path(),
-    }
+def _abs_parts(path):
+    if path.root:
+        return (path.drive, path.root) + path.parts[1:]
+    else:
+        return path.parts
+
+def _file_relative_to(path, base):
+    """
+    Like Path.relative_to, except:
+
+    - `base` must be a single Path object.
+    - The error message is blank.
+    - Only a tuple of parts is returned.
+
+    Surprisingly, this is much, much faster.
+    """
+    child_parts = _abs_parts(path)
+    base_parts = _abs_parts(base)
+
+    n = len(base_parts)
+    cf = path._flavour.casefold_parts
+
+    if n == 0:
+        compare = (path.root or path.drive)
+    else:
+        compare = cf(child_parts[:n])
+
+    if compare != cf(base_parts):
+        raise ValueError()
```

Comment:
The error message being empty has a weird effect on reading.

The first thing you think about is that we might as well return `None` and skip exception handling overhead if we don't care about the error. However, it's not caught in `ResourceRoot.file_to_resource_path` and instead caught in `ResourcePath.from_file_path` and ignored to continue the loop.

From how this pattern is used, I would suggest not raising an exception that nobody will care about and just return `None` and adjusting code paths to care for that (notably `ResourceRoot.file_to_resource_path` and its callee). Should increase performance even more.

Reply:
Done.